core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleIRI.java

```diff
@@ -102,4 +102,13 @@ public String getLocalName() {
 		return iriString.substring(localNameIdx);
 	}
 
+	//conversion function to decode percent-encoded chars, Issue: 1291
+	@Override
+	public IRI percentageDecode(IRI url){
+		String url_ = url.toString();
+		url_ = url_.replaceAll("%", "");
+		url = Values.iri(url_);
+        return url;
+	}
```

Comment:
This looks like it may have accidentally ended up in this PR - I don't think this has anything to do with #3681?